python/ql/src/experimental/dataflow/internal/DataFlowPublic.qll

```diff
@@ -20,7 +20,7 @@ newtype TNode =
   /** A node corresponding to an SSA variable. */
   TEssaNode(EssaVariable var) or
   /** A node corresponding to a control flow node. */
-  TCfgNode(ControlFlowNode node)
+  TCfgNode(DataFlowCfgNode node)
```

Comment:
We have quite a few indirections (`TCfgNode` :arrow_right: `DataFlowCfgNode` :arrow_right: `isExpressionNode`) until we get to the actual definition of what kinds of control flow nodes we want in our set of data flow nodes, and I'm wondering if we could make do with less?

Why not just `TCfgNode(ControlFlowNode n) { n.getNode() instanceof Expr }`?

Reply:
I wanted `isExpressionNode` available for the ([required](https://github.com/github/codeql/blob/main/docs/ql-libraries/dataflow/dataflow.md#node-subclasses)) definition of `ExprNode`.

Reply:
Perhaps we should add comments stating which classes and predicates are required by the library...there is no formal interface mechanism, code just fails to compile if you miss a definition..